warehouse/query-core/src/test/java/datawave/query/LongRunningQueryTest.java

```diff
@@ -0,0 +1,306 @@
+        logic.setIncludeDataTypeAsField(true);
+        logic.setMarkingFunctions(new MarkingFunctions.Default());
+        logic.setMetadataHelperFactory(new MetadataHelperFactory());
+        logic.setDateIndexHelperFactory(new DateIndexHelperFactory());
+        logic.setResponseObjectFactory(new DefaultResponseObjectFactory());
+        // this parameter is what makes the query long running. Failing to set this will let it default to 50 minutes
+        // (and not the 10 milliseconds that it is set to) Set to just 10 ms because since there will be no results,
+        // we have to make sure we hit the timeout before no results are returned.
+        logic.setQueryExecutionForPageTimeout(10);
+        logic.setLongRunningQuery(true);
+        
+        GenericQueryConfiguration config = logic.initialize(connector, query, Collections.singleton(auths));
+        logic.setupQuery(config);
+        
+        RunningQuery runningQuery = new RunningQuery(connector, AccumuloConnectionFactory.Priority.NORMAL, logic, query, "", datawavePrincipal,
+                        new QueryMetricFactoryImpl());
+        List<ResultsPage> pages = new ArrayList<>();
+        
+        ResultsPage page = runningQuery.next();
+        pages.add(page);
+        
+        // guarantee the need for at least a second page.
+        try {
+            Thread.sleep(250);
+        } catch (InterruptedException e) {
+            
+        }
+        
+        while (page.getStatus() != ResultsPage.Status.NONE) {
+            page = runningQuery.next();
+            pages.add(page);
+        }
+        
+        // There should be at least 2 pages, more depending on cpu speed.
+        assertTrue(pages.size() > 1);
+        for (int i = 0; i < pages.size() - 1; ++i) {
+            // check every page but the last one for 0 results and PARTIAL status
+            assertEquals(0, pages.get(i).getResults().size());
+            assertEquals(ResultsPage.Status.PARTIAL, pages.get(i).getStatus());
+        }
+        // check the last page for COMPLETE status and that the total number of results is 8
+        assertEquals(0, pages.get(pages.size() - 1).getResults().size());
+        assertEquals(ResultsPage.Status.NONE, pages.get(pages.size() - 1).getStatus());
+    }
+    
+    /**
+     * Tests that having results returned over multiple pages (by setting the page size to be less than the number of results that will be returned) behaves
+     * even with long running queries exceeding their execution timeout for a page. Should get at least 1 page of 0 results and a PARTIAL status, but could be
+     * more depending on cpu speed due hitting the execution timeout. The last two pages should have 4 results each, but the last page should have a status of
+     * COMPLETE, and the next to last page should have a status of PARTIAL.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testAllowLongRunningQueryWithSmallPageSize() throws Exception {
```

Comment:
When I run this test by itself in my IDE it hangs indefinitely.  It runs fine when running all of the tests in the suite though, which is odd.  Debugging it, it looks like it's getting hung up in the next page while loop.  Probably worth figuring out what's going on here.

Reply:
Try again - a new version has been pushed